app/services/messages/service.go

```diff
@@ -90,17 +90,26 @@ func (ss *Service) SanityCheckSignature(tm encoding.TopicMessage) (bool, error)
 		ss.logger.Errorf("Failed to retrieve Transaction Record for TX ID [%s]. Error: %s", topicMessage.TransactionId, err)
 		return false, err
 	}
+
+	valid, _ := ss.contractsService.IsValidBridgeAsset(t.Asset)
+	if !valid {
+		ss.logger.Errorf("Provided Asset is not supported - [%s]", t.Asset)
+		return false, err
+	}
+
 	match := t.EthAddress == topicMessage.EthAddress &&
 		t.Amount == topicMessage.Amount &&
-		t.Fee == topicMessage.Fee
+		t.Fee == topicMessage.Fee &&
+		t.GasPriceWei == topicMessage.GasPriceWei
+		// TODO: Add this line when functionality supports it -> erc20Address == topicMessage.Erc20Address
```

Comment:
I think that we could add it now. The topic message contains the raw data (`ethaddress, amount, fee, gasprice, erc20address`) and the `signature`.
We could add the `erc20Address` right now and submit it as part of the topic. its just that we are not going to sign it as part of the authorisation signature yet